### PR TITLE
DOC: update docs on building with Meson

### PR DESCRIPTION
**Pull request by rgommers**
[skip azp] [skip github]

### CODE REVIEW

File: doc/source/dev/contributor/meson.rst [hunk 1]
```diff
@@ -6,21 +6,17 @@ How to build SciPy with Meson
...
+   The change over from a `numpy.distutils` to a Meson based build happened
+   recently, so there may still be rough edges. All platforms are supported,
+   however niche configurations may not yet be tested.  These instructions
+   should work reliably on Linux and macOS with a conda environment and
+   OpenBLAS as the BLAS/LAPACK library.
+   Building SciPy on Windows has always been a little tricky - it does work
+   provided that all build dependencies are installed (most importantly,
+   OpenBLAS and a compatible set of compilers: either Mingw-w64 or MSVC + Intel
+   Fortran works.  If you have issues on Windows, please look at
+   `this CI job <https://github.com/rgommers/scipy/blob/meson/.github/workflows/windows.yml>`_
```
[the remark this answers is not in the document]
Comment (rgommers) [reply doc/source/dev/contributor/meson.rst:18]: Good catch, I missed that. No, nothing needs to point at my fork anymore. I'll update.



> The other bizarre thing is what happens if you try to use `dev.py` to build on Windows and you don't have `meson` installed--the error message is almost impossible to decipher, but I can open a separate issue for that if we care.

yes please, that should be easy to improve

> The biggest pain point I saw, when pulling in "just any OpenBLAS" from i.e., `conda` was related to linking UCRT I think.

I'm not aware of the dev workflow not working with `conda` dependencies on Windows. Would be nice if you could open an issue.

File: doc/source/dev/contributor/meson.rst [hunk 2]
```diff
@@ -6,21 +6,17 @@ How to build SciPy with Meson
...
+   recently, so there may still be rough edges. All platforms are supported,
+   however niche configurations may not yet be tested.  These instructions
+   should work reliably on Linux and macOS with a conda environment and
+   OpenBLAS as the BLAS/LAPACK library.
+   Building SciPy on Windows has always been a little tricky - it does work
+   provided that all build dependencies are installed (most importantly,
+   OpenBLAS and a compatible set of compilers: either Mingw-w64 or MSVC + Intel
+   Fortran works.  If you have issues on Windows, please look at
+   `this CI job <https://github.com/scipy/scipy/blob/main/.github/workflows/windows.yml>`_
+   for details.
```
Comment (h-vetinari) [inline doc/source/dev/contributor/meson.rst:19]: Bracket is never closed.

Comment (rgommers) [reply doc/source/dev/contributor/meson.rst]: thanks, fixed

File: doc/source/dev/contributor/meson.rst [hunk 3]
```diff
@@ -187,15 +175,15 @@ it does mean that one current workflow is no longer supported.
...
 If you use an IDE with, e.g., a "Run" button for scripts which were pointing to
 an in-place build, and you would really like to continue using that same
 workflow instead of ``python dev.py``, then you have a few options:
 
 - After modifying pure Python code in the SciPy repo, install it on the command
-  line with ``python dev.py --only-build``, or with ``meson install -C build``
+  line with ``python dev.py build``, or with ``meson install -C build``
   before running your script.
 - If your IDE supports it, customize what the "Run" button does before running
   the script, to do the install each time (this is expected to take 2-3 sec.)
```
Comment (tupui) [inline doc/source/dev/contributor/meson.rst:189]: This only works if you export the PYTHONPATH somehow. Until we have in-place build, we should add a note about this.

**Review by tupui — COMMENTED** [review]
Thanks Ralf. A few suggestions for me.

File: doc/source/dev/contributor/meson.rst (hunk 3 above)
Comment (rgommers) [reply doc/source/dev/contributor/meson.rst]: added a note on this

File: doc/source/dev/contributor/meson.rst [hunk 4]
```diff
@@ -187,21 +175,23 @@ it does mean that one current workflow is no longer supported.
...
-  line with ``python dev.py --only-build``, or with ``meson install -C build``
+  line with ``python dev.py build``, or with ``meson install -C build``
   before running your script.
 - If your IDE supports it, customize what the "Run" button does before running
-  the script, to do the install each time (this is expected to take 2-3 sec.)
-  before executing the script. *Note that the Spyder IDE does not yet support
-  this; its developers are looking at implementing support before the SciPy
-  1.9.0 release).*
+  the script, to do the install each time (this is expected to take 2-3 sec)
+  before executing the script. For this to work, the install directory
```
Comment (tupui) [inline doc/source/dev/contributor/meson.rst:190]:
```suggestion
  For this to work, the install directory
```

**Review by tupui — APPROVED** [review]
LGTM aside from this last thing.

**Comment by rgommers** [comment]
Thanks for the reviews @tupui. I'll squash-merge this now then to stop the CI runs.

**Pull request merged**